OfficeDev/Microsoft-Teams-Samples
# App SSO NodeJS

Description
This sample app illustrates Bot Tab ME SSO Scenario.

Language Used: javascript/node js

Proposed features
Teams Tab, Bot, ME - search, action, linkunfurl SSO with Node JS

Testing
You can interact with this bot by sending it a message. The bot will respond by requesting you to login to AAD, then making a call to the Graph API on your behalf and returning the results.
- Install App

![image](https://user-images.githubusercontent.com/85108465/121664976-acb76400-cac5-11eb-8256-53a8d7980476.png)

- Type *anything* on the compose box and send
- The bot will perform `Single Sign-On` and Profile card will be displayed along with the option prompt to view the `token`

![image](https://user-images.githubusercontent.com/85108465/122400808-ae879880-cf99-11eb-8cc2-5da9b8e420ca.png)

> **NOTE:** 
If the user is using the application for the first time and user consent is required for additional permissions, the following dialog box appears to continue with the consent experience
![image](https://user-images.githubusercontent.com/85108465/122513983-c3f7d380-d028-11eb-913c-1794c5ff851a.png)
![image](https://user-images.githubusercontent.com/85108465/122557279-3503af00-d05a-11eb-981d-bf8db77ff2ac.png)

>If the bot couldn't perform `SSO` then it will fallback to normal Authentication method and show a `Sign In` card like below
![image](https://user-images.githubusercontent.com/85108465/122401855-9a906680-cf9a-11eb-94f1-87840c6662b4.png)

- Open `Messaging Extension`(Search), it will show profile details

![image](https://user-images.githubusercontent.com/85108465/121668748-3ddc0a00-cac9-11eb-8c0e-cc3d60f2b5a8.png)
![image](https://user-images.githubusercontent.com/85108465/121669133-a32ffb00-cac9-11eb-9427-50321a317550.png)
![image](https://user-images.githubusercontent.com/85108465/121669220-bc38ac00-cac9-11eb-982e-880ebb8258a8.png)

- Open `Messaging Extension`(Action), it will show profile details

![image](https://user-images.githubusercontent.com/85108465/121669676-2f422280-caca-11eb-8659-3f8f08d11bbd.png)

__*or*__

![image](https://user-images.githubusercontent.com/85108465/121669730-3e28d500-caca-11eb-9ba5-fca1b97e7f20.png)
![image](https://user-images.githubusercontent.com/85108465/121669695-32d5a980-caca-11eb-9893-2d3a3a3b1821.png)

- Open `Messaging Extension`(linkunfurl), The link will unfurl and show profile details

**Paste** https://profile.botframework.com on the compose box

![image](https://user-images.githubusercontent.com/85108465/121669972-93fd7d00-caca-11eb-87bb-e07e0e7aa5e4.png)
![image](https://user-images.githubusercontent.com/85108465/121669990-98c23100-caca-11eb-9a31-30c3d5065853.png)

> NOTE: If `SSO` couldn't be performed then it will fallback to normal Authentication method and you will get a default `Sign In` action

Consent the *ME Search* by clicking the `Sign In` link like below 

![image](https://user-images.githubusercontent.com/85108465/121671255-f2772b00-cacb-11eb-9321-1317696eaccc.png)

Consent the *ME Action* by clicking the `Setup` button like below 

![image](https://user-images.githubusercontent.com/85108465/122556633-6039ce80-d059-11eb-9fa5-e0fe4db4939d.png)

- Open `SSO Tab`, Continue and then Accept and it'll show the profile details

![image](https://user-images.githubusercontent.com/85108465/121671560-5568c200-cacc-11eb-954b-44155e039915.png)
![image](https://user-images.githubusercontent.com/85108465/121671603-61ed1a80-cacc-11eb-9754-ff0b2aaac671.png)
![image](https://user-images.githubusercontent.com/85108465/121672009-e344ad00-cacc-11eb-860f-c0ed5153edc7.png)
